packages/sql-kysely/src/internal/kysely.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * @since 1.0.0
+ */
+import * as Client from "@effect/sql/Client"
+import * as Effect from "effect/Effect"
+import {
+  AlterTableColumnAlteringBuilder,
+  CreateIndexBuilder,
+  CreateSchemaBuilder,
+  CreateTableBuilder,
+  CreateTypeBuilder,
+  CreateViewBuilder,
+  DeleteQueryBuilder,
+  DropIndexBuilder,
+  DropSchemaBuilder,
+  DropTableBuilder,
+  DropTypeBuilder,
+  DropViewBuilder,
+  InsertQueryBuilder,
+  Kysely,
+  UpdateQueryBuilder,
+  WheneableMergeQueryBuilder
+} from "kysely"
+import type { Dialect } from "kysely"
+import type { EffectKysely } from "../patch.types.js"
+import { effectifyWithExecute, effectifyWithSql, patch } from "./patch.js"
+
+/**
+ * @internal
+ * patch all compilable/executable builders with commit prototypes
+ *
+ * @warning side effect
+ */
+patch(AlterTableColumnAlteringBuilder.prototype)
```

Comment:
Hmm this is even more brittle than the drizzle integration. I think we will need to find a more general interface to target if possible.

Reply:
No Luck here. There is no common impl to rely on.

Reply:
When I get some time I'll take a look and see if there is anything we can do.

Reply:
This might have some merit:

```ts
import {
  CompiledQuery,
  DefaultQueryCompiler,
  Kysely,
  SqliteDialect,
} from "kysely"
import { Effect, Effectable } from "effect"

declare module "kysely" {
  export interface CompiledQuery<O = unknown> extends Effect.Effect<O> {}
}

class EffectCompiledQuery<O = unknown>
  extends Effectable.Class<O>
  implements CompiledQuery<O>
{
  constructor(private compiledQuery: CompiledQuery<O>) {
    super()
  }
  commit(): Effect.Effect<O> {
    return Effect.succeed({} as any)
  }
  get sql() {
    return this.compiledQuery.sql
  }
  get query() {
    return this.compiledQuery.query
  }
  get parameters() {
    return this.compiledQuery.parameters
  }
}

Effect.gen(function* () {
  const oldCompileQuery = DefaultQueryCompiler.prototype.compileQuery
  DefaultQueryCompiler.prototype.compileQuery = function (query) {
    console.log("compiling query", query)
    return new EffectCompiledQuery(oldCompileQuery.call(this, query))
  }

  const dialect = new SqliteDialect({} as any)
  const db = new Kysely<any>({
    dialect,
  })
  const result = yield* db.selectFrom("person").where("id", "=", 1).compile()
  console.log(result)
}).pipe(Effect.runPromise)
```

Reply:
My guts tell me that it's not a good enough tradeoff.
it's trading users DX for internal code simplification.
And it don't allow for using native kysely drivers in an Effect integrated way.

Reply:
yeah I'd also try to optimise for usage rather than internal complexity, only wondering if those internals are stable or they are up for change in kysley, if the latter it might be risky to depend on those. I guess we can involve someone from the kysley team to clarify?

Reply:
@igalklebanov your feedback would be valuable here.

Reply:
Hey 👋

Don't have enough context and understanding of "the Effect way".

The main builders are pretty stable.
`SelectQueryBuilder` will *probably* be exposed again.
Builders do get renamed/split on rare ocassions.
Builders are added sometimes, and it's not always a breaking change 

There is no indication `

Reply:
Thanks,

One thing that could simplify things on our side would be to had a Executable class like this :
```ts
class Executable<0> {
  abstract execute() : promise<O>
}
```
And let all builders implement it. This way we would only need to patch this one. And it would be future proof for us. Meaning adding / splitting / removing builders would have no impact on us.
 Would such addition be welcomed in kysely ? Or at least start a discussion on kysely repo about it?

Reply:
I got feedback from kysely team, and my proposal is not something they want to add in their code base.
So either we maintain on our side the compatibility with builders evolution or we can close this PR if it's too much of a concern.

Reply:
There are more ways to execute a query:

`Kysely`'s `executeQuery` method.
`SelectQueryBuilder`, `UpdateQueryBuilder`, `InsertQueryBuilder` and `DeleteQueryBuilder`'s `stream` method (anything that implements `Streamable<O>`).
`sql` template tag's `execute`.

Regarding more future-proof patching:

Have you tried patching `DefaultQueryExecutor`'s `compileQuery`, `executeQuery` and `stream`? It extends `QueryExecutorBase` a class that'll never be exposed, but these parts are 1-2 years untouched and central to execution flow.

Reply:
Unfortunately, Streamable is just an interface, not a class, so prototype is not patchable.
We need a base class to patch and no Builder has a base class.

Reply:
I first read it wrong. Sorry for last message (removed).
So DefaultQueryExecutor is used has the executor for all execute calls ?

Reply:
Just digged deeper into the code, unfortunately, Patching DefaultQueryExecutor would mean changing behaviour of you `execute()` for all instances to return an effect instead of a Promise even when user don't want to use an Effect.
We really need a way to Patch the Builder itself.

Reply:
I think we can keep current solution and see, if the kysley team has no interest to facilitate integration we are left with two choices:
1) trust they won't change internals and eventually adapt for the changes when they come, we need to make sure using a builder that is not supported will raise an error
2) fork kysley and maintain an effect-first fork

I'd start with 1 and get to 2 only if 1 becomes a pain